pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1202,12 +1203,13 @@ func (ctrl *PersistentVolumeController) doDeleteVolume(volume *v1.PersistentVolu
 		// Cannot create deleter
 		return false, fmt.Errorf("Failed to create deleter for volume %q: %v", volume.Name, err)
 	}
-
+	start := time.Now()
 	if err = deleter.Delete(); err != nil {
 		// Deleter failed
+		metrics.VolumeDeleteLatency.Observe(metrics.SinceInMilliseconds(start))
 		return false, err
 	}
-
+	metrics.VolumeDeleteLatency.Observe(metrics.SinceInMilliseconds(start))
```

Comment:
It might be useful to have some dimensions on these metrics such as `volumeName` -- so as we can identify if a particular volume took longer than average to delete, then which volume was it.

These metrics can still be aggregated discarding the dimensions but it is useful when trying to identify problems.